### ISSUE TITLE
GHC 8.10 support

### ISSUE THREAD

**Issue by kozross** [composewell/streamly#463]
Currently, ``ghc-prim`` is version-blocking ``streamly`` from building:

```
[koz@Sebastian streamly-fsnotify]$ cabal new-configure -w /home/koz/ghc-810/bin/ghc
'cabal.project.local' file already exists. Now overwriting it.
Resolving dependencies...
cabal: Could not resolve dependencies:
[__0] trying: streamly-fsnotify-1.0.0.2 (user goal)
[__1] trying: streamly-0.7.1 (dependency of streamly-fsnotify)
[__2] next goal: base (dependency of streamly-fsnotify)
[__2] rejecting: base-4.14.0.0/installed-4.1... (conflict: streamly =>
ghc-prim>=0.2 && <0.6, base => ghc-prim==0.6.1/installed-0.6...)
[__2] rejecting: base-4.12.0.0, base-4.11.1.0, base-4.11.0.0, base-4.10.1.0,
base-4.10.0.0, base-4.9.1.0, base-4.9.0.0, base-4.8.2.0, base-4.8.1.0,
base-4.8.0.0, base-4.7.0.2, base-4.7.0.1, base-4.7.0.0, base-4.6.0.1,
base-4.6.0.0, base-4.5.1.0, base-4.5.0.0, base-4.4.1.0, base-4.4.0.0,
base-4.3.1.0, base-4.3.0.0, base-4.2.0.2, base-4.2.0.1, base-4.2.0.0,
base-4.1.0.0, base-4.0.0.0, base-3.0.3.2, base-3.0.3.1 (constraint from
non-upgradeable package requires installed instance)
[__2] fail (backjumping, conflict set: base, streamly, streamly-fsnotify)
After searching the rest of the dependency tree exhaustively, these were the
goals I've had most trouble fulfilling: base, streamly-fsnotify, streamly
```

**Comment by adithyaov** [comment]
@kozross Thank you for bringing this up. Given the changelog of `ghc-prim`, additional functionality has been added in `ghc-prim-0.6.0` and `ghc-prim-0.6.1`. Nothing seems to have been removed or changed. I think it should be safe to bump up `ghc-prim`.

**Comment by pranaysashank** [comment]
streamly-0.7.2 is now released with GHC-8.10 support.
